File: Lesson13/src/Codes.java
enum Codes {
    EXIT(0), PRINT(1), EDIT(2), SORT(3), ADD(4),DELETE(5);

    public int getCode() {
        return code;
    }

    Codes(int code) {
        this.code = code;
    }

    public static Codes getEnumByCode(int code) throws Exception {
        for (Codes c : Codes.values()) {
            if (c.code == code) {
                return c;
            }
        }
        throw new Exception("No equals");
    }

    private final int code;
}

File: Lesson14/src/Codes.kt
enum class Codes(val code: Int) {
    SHOW(0),
    TRANSLATE(1),
    ADD(2),
    DELETE(3),
    EXIT(4),
    RU(5),
    EN(5),
    ALL(6);

    companion object {
        @Throws(Exception::class)
        fun getEnumByCode(code: Int): Codes {
            for (c in values()) {
                if (c.code == code) {
                    return c
                }
            }
            throw Exception("No equals")
        }
    }
}
